gaek/ndb_json.py

```diff
@@ -109,28 +134,52 @@ def encode_basevalue(obj):
   ndb.MetaModel: encode_model,
   ndb.Query: encode_generator,
   ndb.QueryIterator: encode_generator,
-  ndb.Key: encode_key,
+  ndb.Key: encode_key_as_entity,
   ndb.Future: encode_future,
   datetime.date: encode_datetime,
   datetime.datetime: encode_datetime,
   time.struct_time: encode_generator,
   types.ComplexType: encode_complex,
   ndb.model._BaseValue: encode_basevalue,
-  
 }
 
 
 class NdbEncoder(json.JSONEncoder):
   """Extend the JSON encoder to add support for NDB Models."""
 
+  _ndb_type_encoding=NDB_TYPE_ENCODING.copy()
+
+  def __init__(self, **kwargs):
+    ndb_keys_as_entities = kwargs.pop('ndb_keys_as_entities', False)
```

Comment:
Optional: Feel free to omit the `ndb_` prefix for the internal variable names since it's mostly designed to clarify between the usage of the `json` module's default parameters.